src/index.mjs

```diff
@@ -62,44 +62,35 @@ export function respondCORS() {
   });
 }
 
-function randomPackageRegistry() {
-  return PACKAGE_REGISTRIES[Math.floor(Math.random() * (PACKAGE_REGISTRIES.length))];
-}
-
-export function getOtherPackageRegistry(regName) {
-  return PACKAGE_REGISTRIES[(PACKAGE_REGISTRIES.length - 1) - PACKAGE_REGISTRIES.indexOf(regName)];
-}
-
 async function respondRegistry(regName, req) {
-  console.log('Using package registry', regName);
+  return new Promise((resolve, reject) => {
+    console.log('Using package registry', regName);
 
-  if (regName === 'jsdelivr') {
-    return respondJsdelivr(req);
-  }
-
-  return respondUnpkg(req);
+    try {
+      const respondFunc = regName === 'jsdelivr' ? respondJsdelivr : respondUnpkg;
+      respondFunc(req).then(
+        (resp) => {
+          if (resp.status !== 200) {
+            // Hold off resolving the promise with an error for a few seconds
+            setTimeout(() => reject(new Error(`Registry ${regName} returned ${resp.status}`)), REGISTRY_TIMEOUT_MS);
+            return;
+          }
+          resolve(resp);
+        },
+      );
+    } catch (error) {
+      // If there is an error, hold off throwing it to give the other registry a chance
+      setTimeout(() => reject(error), REGISTRY_TIMEOUT_MS);
+    }
+  });
 }
 
 async function respondPackage(req) {
-  let pkgreg = new URL(req.url).searchParams.get('pkgreg');
-
-  if (!pkgreg) {
-    pkgreg = randomPackageRegistry();
-  }
-
-  try {
-    let resp = await respondRegistry(pkgreg, req);
-    if (resp.status !== 200) {
-      console.log('Changing registry as its response was', resp.status);
-      resp = await respondRegistry(getOtherPackageRegistry(pkgreg), req);
-    }
-    return resp;
-  } catch (error) {
-    console.log('Contacting registry caused this error', error);
-    console.log('Changing package registry');
-
-    return respondRegistry(getOtherPackageRegistry(pkgreg), req);
-  }
+  // TODO possibly randomize the order of the array?
+  return /* await */ Promise.race([
```

Comment:
@trieloff wouldn't it be easier to use `Promise.any()` here? 

That 'promise fulfills when any of the input's promises fulfills. It rejects when all of the input's promises reject' - isn't that exactly what we need here?